### DISCUSSION TITLE
mapSeed Detection

### DISCUSSION THREAD

**Discussion by i-caused-y2k** [blacha/diablo2#403] [Q&A] [answered]
Hi @blacha, great work here. I started porting this code for Mac and Windows. However, I keep getting a different mapSeed every time it scans. It does correctly detect the player, act, and difficulty, though.

```typescript
const mapSeed = await obj.d2.readStrutAt(this.mapSeed, bp.lu32);
```

The first time this runs, `this.mapSeed` is `undefined`. I don't see anywhere it could've been set, either. So, I cast that to `0` for the first run. It now returns a large number, with 2 issues: it's different every single time, and every so often (1 of 20 or so?) it returns `0`.

I thought perhaps I could use the `ActMiscStrut` (like in `getDifficulty`)...

```typescript
{
  tombLevel: at(0x120, lu32),
  difficulty: at(0x830, lu16),
  mapSeed: at(0x840, lu32),
  pAct: at(0x858, new Pointer(u8)),
  pLevelFirst: at(0x868, new Pointer(u8)),
}
```

...now I can get a consistent value for `mapSeed`. However, when I put that into your map generator online, the result does not correspond to the actual map in-game.

Any idea how to proceed? Thanks!

**Comment by blacha** [comment, marked as answer]
I havent played D2R in quite a while, Last I heard they had obsfucated the mapseed in memory.

So this logic wil not work with the current build of the game, It is also highly likely that this will now result in a ba as MapAssist users were banned recently.

Here is some examples of finding the mapseed https://github.com/OneXDeveloper/MapAssist/blob/4b56415631bae303a389416fcd59df3605d80def/Types/MapSeed.cs#L6
